engine/src/main/java/org/terasology/engine/subsystem/rpc/DiscordRPCSubSystem.java

```
@@ -0,0 +1,308 @@
+                if (dontTryAgain) {
+                    Thread.sleep(1);
+                    continue;
+                }
+
+                // Connect if the connect on init didn't connect successfully
+                if (!connectedBefore && !ready) {
+                    lastPing = 0;
+                    try {
+                        ipcClient.connect();
+                    } catch (Exception ex) { }
+                    Thread.sleep(15 * 1000);
+                    if (!ready) {
+                        reconnectTries += 1;
+                        if (reconnectTries >= 5) {
+                            dontTryAgain = true;
+                        }
+                    }
+                    continue;
+                }
+
+
+                // Ping to make sure that the RPC is alive
+                if (ready) {
+                    Thread.sleep(1);
+                    lastPing += 1;
+                    if (lastPing >= 5 * 1000) {
+                        ipcClient.sendRichPresence(this.lastRichPresence);
+                        this.lastPing = 0;
+                    }
+                } else { // Reconnect to RPC with tries ( min: 2 seconds, max: 30 seconds )
+                    lastPing = 0;
+                    reconnecting = true;
+                    int timeout = (reconnectTries * 2) * 1000;
+                    getLogger().info("Discord RPC >> Reconnecting... (Timeout: " + timeout + "ms)");
+                    try {
+                        ipcClient.connect();
+                    } catch (Exception ex) {
+                        if (reconnectTries <= 5) {
+                            reconnectTries += 1;
+                        }
+                        if (reconnectTries >= 5) {
+                            dontTryAgain = true;
+                        }
+                        Thread.sleep(timeout);
+                    }
+                }
+            } catch (Exception ex) { }
```

Might want a comment explaining why the exception is ignored 👍

---

**Done** I added a clearly stuff to handle the errors and ignore the `InterruptedException`.